src/arraystring.rs

```diff
@@ -577,10 +472,15 @@ where
     #[inline]
     pub fn try_push_str(&mut self, string: impl AsRef<str>) -> Result<(), OutOfBounds> {
         trace!("Push str: {}", string.as_ref());
-        let new_end = string.as_ref().len().saturating_add(self.len().into());
-        is_inside_boundary(new_end, Self::capacity())?;
-        unsafe { self.push_str_unchecked(string) };
-        Ok(())
+        let str = string.as_ref().as_bytes();
+        if str.len() == 0 {
+            return Ok(());
+        }
+        is_inside_boundary(str.len() + self.len(), Self::capacity())?;
```

Comment:
Cool, now that everything is usize we are garanteed to not wrap. Pretty much even in targets where usize has 16 bits or 32 bits.

Nice to be less verbose than saturating_add everywhere

Reply:
yeah, i was getting annoyed by all the conversions everywhere so i just changed everything to usize.